tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go

```diff
@@ -98,46 +102,41 @@ func TestEgressGatewayTls(t *testing.T) {
 					response:            []string{response.StatusCodeBadRequest},
 					gateway:             false, // 400 response will not contain header
 				},
-				// Egress Gateway can't "originate" ISTIO_MUTUAL to server as the server is outside the mesh and has no
-				// sidecar proxy
-				// 4. ISTIO_MUTUAL TLS origination case::
-				//    internalClient ) ---HTTP request (Host: some-external-site.com----> Hits listener 0.0.0.0_80 ->
-				//      VS Routing (add Egress Header) --> Egress Gateway(originates istio mutual TLS)
-				//      --> externalServer(443 with TLS enforced)
-				//     request fails as server has no sidecar proxy and istio_mutual case shouldn't be used
-				// TODO: nschhina figure out why the following tests are flaky
-				"ISTIO_MUTUAL TLS origination from egress gateway to https endpoint": {
-					destinationRuleMode: "ISTIO_MUTUAL",
-					response:            []string{response.StatusCodeBadRequest},
-					gateway:             false, // 400 response will not contain header
-					fakeRootCert:        false,
-				},
 				//5. SIMPLE TLS origination with "fake" root cert::
 				//   internalClient ) ---HTTP request (Host: some-external-site.com----> Hits listener 0.0.0.0_80 ->
 				//     VS Routing (add Egress Header) --> Egress Gateway(originates simple TLS)
 				//     --> externalServer(443 with TLS enforced)
 				//    request fails as the server cert can't be validated using the fake root cert used during origination
 				"SIMPLE TLS origination from egress gateway to https endpoint with fake root cert": {
 					destinationRuleMode: "SIMPLE",
-					response:            []string{response.StatusCodeBadRequest},
-					gateway:             false, // 400 response will not contain header
+					response:            []string{response.StatusCodeUnavailable},
+					gateway:             false, // 503 response will not contain header
 					fakeRootCert:        true,
 				},
 			}
 
 			for name, tc := range testCases {
 				t.Run(name, func(t *testing.T) {
-					bufDestinationRule := createDestinationRule(t, appNamespace, tc.destinationRuleMode, tc.fakeRootCert)
+					bufDestinationRule := createDestinationRule(t, serviceNamespace, tc.destinationRuleMode, tc.fakeRootCert)
+
+					istioCfg := istio.DefaultConfigOrFail(t, ctx)
+					systemNamespace := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+					// Wait for all CDS configs to update
+					time.Sleep(10 * time.Second)
 
-					ctx.Config().ApplyYAMLOrFail(ctx, appNamespace.Name(), bufDestinationRule.String())
-					defer ctx.Config().DeleteYAMLOrFail(ctx, appNamespace.Name(), bufDestinationRule.String())
+					ctx.Config().ApplyYAMLOrFail(ctx, systemNamespace.Name(), bufDestinationRule.String())
+					defer ctx.Config().DeleteYAMLOrFail(ctx, systemNamespace.Name(), bufDestinationRule.String())
+
+					// Wait for all CDS configs to update
+					time.Sleep(10 * time.Second)
```

Comment:
same here. 20 secs wait in a test is not good. It is better to keeping checking updates periodically and return early.

Reply:
I reduced this wait time, only issue to periodically checking is how to get gateway config dump to check against dynamicactiveclusters. I think new wait time is reasonable if we find no more flakes.